rust/pattern/variable/thing_variable.rs

```diff
@@ -89,41 +109,15 @@ impl RelationConstrainable for ThingVariable {
     }
 }
 
-impl Display for ThingVariable {
+impl fmt::Display for ThingVariable {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        if self.reference.is_visible() {
-            write!(f, "{}", self.reference)?;
-        }
+        self.fmt_thing_syntax(f)?;
 
-        if let Some(value) = &self.value {
-            if self.reference.is_visible() {
-                f.write_char(' ')?;
-            }
-            write!(f, "{}", value)?;
-        }
-        if let Some(relation) = &self.relation {
-            if self.reference.is_visible() {
-                f.write_char(' ')?;
-            }
-            write!(f, "{}", relation)?;
-        }
-
-        if let Some(iid) = &self.iid {
-            write!(f, " {}", iid)?;
+        if self.is_thing_constrained() {
+            f.write_str(" ")?;
+            write_joined!(f, ",\n    ", self.isa, self.iid, self.has)?;
```

Comment:
More powerful `write_joined!` now allows us to string together sequences of different types. (Remember that `Option` is treated as a sequence of zero or one elements)